File: samplejava/src/main/java/com/kazakago/factolize/samplejava/MainFragment.java
package com.kazakago.factolize.samplejava;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import com.kazakago.factolize.Factory;

@Factory
public class MainFragment extends Fragment {

    private EditText intEditText;
    private EditText stringEditText;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MainFragmentFactory.injectArgument(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        intEditText = view.findViewById(R.id.intEditText);
        stringEditText = view.findViewById(R.id.stringEditText);

        Button goToSubActivityButton = view.findViewById(R.id.goToSubActivityButton);
        goToSubActivityButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                goSubActivity();
            }
        });
    }

    private void goSubActivity() {
        int intValue = (!intEditText.getText().toString().isEmpty()) ? Integer.valueOf(intEditText.getText().toString()) : 0;
        String stringValue = stringEditText.getText().toString();

        Intent intent = SubActivityFactory.createIntent(getActivity(), intValue, stringValue);
        startActivity(intent);
    }
}

File: samplekotlin/src/main/java/com/kazakago/factolize/samplekotlin/MainFragment.kt
package com.kazakago.factolize.samplekotlin

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import com.kazakago.factolize.Factory
import kotlinx.android.synthetic.main.fragment_main.*

@Factory
class MainFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        MainFragmentFactory.injectArgument(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val goToSubActivityButton = view.findViewById<Button>(R.id.goToSubActivityButton)
        goToSubActivityButton.setOnClickListener {
            val intValue = intEditText.text.toString().toIntOrNull() ?: 0
            val stringValue = stringEditText.text.toString()

            goSubActivity(intValue, stringValue)
        }
    }

    private fun goSubActivity(intValue: Int, stringValue: String) {
        val intent = SubActivityFactory.createIntent(activity!!, intValue, stringValue)
        startActivity(intent)
    }

}
